Update ballerina command to bal in Using the CLI Tools of Learn pages (#1705)
ballerina-platform/ballerina-dev-website: swan-lake/learn/using-the-cli-tools.md

---
layout: ballerina-left-nav-pages-swanlake
title: Using the CLI Tools

> description: Learn all the command line interface (CLI) commands need to get started, build, test and run programs, work with Ballerina Central, and manage packages.

Reviewer: ```suggestion
description: Learn all the command line interface (CLI) commands you need to get started, build, test and run programs, work with Ballerina Central, and manage packages.
```

Author: Fixed.

## Using the Ballerina Tool

The Ballerina Tool is a command-line tool for managing Ballerina source code. It helps you to manage Ballerina packages and modules, test, build, and run programs, etc.

It also enables you to easily install, update, and switch among Ballerina distributions. 

In the CLI, execute the `bal help` command to view all the actions you can perform with the Ballerina Tool as shown below:

```sh
→ bal help
NAME

>        The bal build tool

Reviewer 2: I think we should leave the name as `Ballerina`, just the command usages should be changed to `bal`.

Author: This is an output generated by the `bal` CLI. 

@keizer619 can this helper text be changed?

Author: Will be address at https://github.com/ballerina-platform/ballerina-lang/issues/28138

SYNOPSIS
       bal <-v | --version>
       bal [command] <-h | --help>
       bal <command> [<args>]

DESCRIPTION

>        Ballerina is a statically typed, concurrent programming language, focusing on network interaction and structured

Reviewer: ```suggestion
       Ballerina is a statically-typed, concurrent programming language focusing on network interaction and structured
```

Author: @praneesha this as well is a CLI output generated by the `bal help` command.
Need to specifically address these suggested CLI output text issues in the [ballerina-lang repo](https://github.com/ballerina-platform/ballerina-lang).

Author: Will be address at https://github.com/ballerina-platform/ballerina-lang/issues/28138

>        Ballerina is a statically typed, concurrent programming language, focusing on network interaction and structured

Reviewer: ```suggestion
       Ballerina is a statically-typed, concurrent programming language, focusing on network interaction and structured
```

Author: This is the CLI output of the `bal help` command.

@keizer619 WDYT?

Author: Will be address at https://github.com/ballerina-platform/ballerina-lang/issues/28138

>        Ballerina is a statically typed, concurrent programming language, focusing on network interaction and structured

Reviewer: ```suggestion
       Ballerina is a statically-typed, concurrent programming language focusing on network interaction and structured
```

Author: An issue was created [here](https://github.com/ballerina-platform/ballerina-lang/issues/28138) regarding [this](https://github.com/ballerina-platform/ballerina-dev-website/pull/1705#discussion_r562591836) comment.

It also enables you to easily install, update, and switch among Ballerina distributions. 

In the CLI, execute the `bal help` command to view all the actions you can perform with the Ballerina Tool as shown below:

```sh

> → ballerina help 

Reviewer 3: ```suggestion
→ bal help 
```

Author: Fixed.

   Core Commands:
        build           Compile Ballerina program into an executable
        run             Build and run Ballerina program
        test            Run package tests
        doc             Generate API documentation
        clean           Clean artifacts generated during the build
        format          Format Ballerina sources

   Package Commands:
        new             Create a new Ballerina package

>         init            Create a new Ballerina package in the current directory

Reviewer 2: Can we use the word initialize here instead of create?

Author: This is a CLI output generated by the `bal help` tool. 
@praneesha WDYT about marking this as an improvement issue in the [ballerina-lang repo](https://github.com/ballerina-platform/ballerina-lang)?

Reviewer: @keizer619 - WDYT?

Author: Will be address at https://github.com/ballerina-platform/ballerina-lang/issues/28138

>         add             Create a new Ballerina module in the package

Reviewer 2: Let's use the word **add** instead of **create**.

Author: This is also a CLI output generated by the `bal help` tool. 
@praneesha WDYT about marking this as an improvement issue in the [ballerina-lang repo](https://github.com/ballerina-platform/ballerina-lang)?

Reviewer: I am fine with what we have now also. @keizer619 - WDYT?

Author: Will be address at https://github.com/ballerina-platform/ballerina-lang/issues/28138

   Update Commands:
        dist            Manage Ballerina distributions
        update          Update the Ballerina tool

Use 'bal help <command>' for more information on a specific command.
```

You can use it in the below format.

> `bal <THE-COMMAND> <ITS-ARGUEMENTS>`

> > **Tip:** You can view details of the commands below by executing `bal help <COMMAND>`. For example, the below is the output of the `bal help pull` command.

Reviewer: ```suggestion
> **Tip:** You can view details of the commands below by executing the `bal help <COMMAND>`. For example, the below is the output of the `bal help pull` command.
```

Author: Fixed

<table class="cComandTable">
<tr>
<td class="cCommand">build</td>

> <td class="cDescription">Compile a Ballerina program, a single `.bal` file or an entire package into an executable JAR file. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.

Reviewer 3: ```suggestion
<td class="cDescription">Compile a standalone `.bal` file or an entire package into an executable JAR file. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.
```

Author: Fixed.

> <td class="cDescription">Compile a Ballerina program, a single `.bal` file or an entire package into an executable JAR file. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.

Reviewer: ```suggestion
<td class="cDescription">Compile a Ballerina program, a single `.bal` file, or an entire package into an executable JAR file. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.
```

Author: Fixed with [this](https://github.com/ballerina-platform/ballerina-dev-website/pull/1705#discussion_r563471975) suggestion as well.

> <td class="cDescription">Compile a Ballerina program, a single BAL file, a module, an entire package, or a single root module into an executable JAR file. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.

Reviewer 4: We can't build modules. Also, root module is the package.
```suggestion
<td class="cDescription">Compile a Ballerina program, a single `.bal` file or an entire package into an executable JAR file. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.
```

Author: Fixed.

</td>
</tr>
<tr>
<td class="cCommand">run</td>

> <td class="cDescription">Build and run a Ballerina program, a single `.bal` file, an entire package, or a previously-built program. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.

Reviewer 3: ```suggestion
<td class="cDescription">Build and run a standalone `.bal` file, an entire package, or a previously-built program. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.
```

Author: Fixed.

> <td class="cDescription">Build and run a Ballerina program, a single BAL file, an entire package, or a previously-built program. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.

Reviewer 4: ```suggestion
<td class="cDescription">Build and run a Ballerina program, a single `.bal` file, an entire package, or a previously-built program. For more information, see <a href="/swan-lake/learn/running-ballerina-code">Running Ballerina Code</a>.
```

Author: Fixed.

</td>
</tr>
<tr>
<td class="cCommand">test</td>

> <td class="cDescription">Run tests of a particular module or all the modules of a Ballerina package. For more information, see <a href="/swan-lake/learn/testing-ballerina-code/testing-quick-start/">Testing Ballerina Code</a>.

Reviewer 4: can't run tests for a particular module right?

Author: @azinneera WDYT?

Author: @rdulmina the `bal help` command states that `Run package tests`.
I'll be updating this to `Run tests of a Ballerina package`.

Author: Updated.

</td>
</tr>
<tr>
<td class="cCommand">doc</td>

> <td class="cDescription">Generate API documents for all public symbols of a Ballerina module or package. For more information, see <a href="/swan-lake/learn/documenting-ballerina-code">Documenting Ballerina Code</a>.

Reviewer 4: Only for package

Author: @rdulmina the output of the `bal help doc` is as following: 
![image](https://user-images.githubusercontent.com/9464997/105646716-763acd00-5ec7-11eb-8cf0-5b5c79056e75.png)
According to the provided description, the `bal doc` is used to build the documentation for a Ballerina package.

Reviewer 4: Yes, then we should change `Generate API documents for all public symbols of a Ballerina module or package` to `Generate API documents for all public symbols of a Ballerina package`.

Author: Thanks, updated.

</tr>
</table>

## Package Commands

> Ballerina packages are the way to organize real-world Ballerina development tasks. Some of them also allow you to work with the Ballerina Central to share Ballerina packages with others in a safe, secure, and dependable way.

Reviewer: ```suggestion
Ballerina packages are the way to organize real-world Ballerina development tasks. Some of them allow you to work with the Ballerina Central also to share Ballerina packages with others in a safe, secure, and dependable way.
```

Author: Fixed.

> Ballerina packages are the way to organize real-world Ballerina development tasks. Some of them allow you to work with the Ballerina Central also to share Ballerina packages with others in a safe, secure, and dependable way.

Reviewer 2: ```suggestion
Ballerina packages are the way to organize real-world Ballerina development tasks. The last 3 commands given below allow you to work with the Ballerina Central and also to share Ballerina packages with others in a safe, secure, and dependable way.
```

Reviewer: We do not need to mention the specifics in the intro.

Reviewer: +1

Author: Fixed.

</tr>
</table>

## Package Commands

Ballerina packages are the way to organize real-world Ballerina development tasks. Some of them also allow you to work with the Ballerina Central to share Ballerina modules with others in a safe, secure, and dependable way.

<table class="cComandTable">
<tr>
<td class="cCommand">new</td>
<td class="cDescription">Create a Ballerina package. For more information, see <a href="/swan-lake/learn/structuring
-ballerina-code">Structuring Ballerina Code</a>.
</td>
</tr>
<tr>
<td class="cCommand">init</td>
<td class="cDescription">Create a new Ballerina package in the current directory.
</td>
</tr>
<tr>
<td class="cCommand">add</td>
<td class="cDescription">Create a new Ballerina module in a package.
</td>
</tr>
<tr>
<td class="cCommand">pull</td>
<td class="cDescription">Pull a module from Ballerina Central.
</td>
</tr>
<tr>
<td class="cCommand">push</td>

> <td class="cDescription">Upload a module to Ballerina Central. For more information, see <a href="/swan-lake/learn/publishing-modules-to-ballerina-central">Publishing Modules to Ballerina Central</a>.

Reviewer 4: Ballerina Central works with packages, not modules. Please change the other places accordingly.

Reviewer 4: No need to change the url it will be changed in [this PR](https://github.com/ballerina-platform/ballerina-dev-website/pull/1714).

Author: Fixed.

</tr>
<tr>
<td class="cCommand">help</td>

> <td class="cDescription">Prints the usage details of any Ballerina command (e.g., `bal help pull`).

Author: @azinneera will it be okay to list the `help` command description under the **Other Commands** section?

Reviewer: +1
